text/0208-ephemeral-resources.md

```diff
@@ -0,0 +1,78 @@
+# Ephemeral Resource Attributes
+
+Define a new type of resource attribute, ephemeral resources, which are allowed to change over the lifetime of the process. Existing resources are redefined as permanent resources, which must be present at SDK initialization and cannot be changed.
```

Comment:
I have proposed a somewhat similar OTEP https://github.com/open-telemetry/oteps/pull/207

If #207 was implemented you could store your ephemeral resource attributes on the Context, and replace the active context when they change. Please check if #207 would also cover your use case.

Reply:
Thanks! That looks like a good proposal, but the context scope still presumes a transactional scope within a server handling many independent transactions. 

For clients, all telemetry emitted, including logs which are not bounded by a span, are related. Which is why the resource scope appears to be the correct one for things like this.

Reply:
That might work... but it might be better to keep the concept of a "process scope" and a "context scope" separate. I see these attributes as more similar to resources and instrumentation scopes - they represent the environment the transaction is occurring within. 

Because contexts are immutable, and no rules as to when child contexts may be created, there would be synchronization issues between when ephemeral resources are updated and when they would applied, if they only change the root context and thus only affect transactions which start from a new root context.